### PR TITLE
Add glossary, fix small typo

### PR DESCRIPTION
**Pull request by evemorgen**
This PR brings user friendly glossary

### CODE REVIEW

File: docs/user/drivers/index.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ It is easy to change some logic by writing your own driver and using it.
 Entry points
 ------------
 
-To make it easier to create and use other drivers without code changes within VmShepherd itself, python's entry point mechanism are used (https://packaging.python.org/specifications/entry-points/).
+To make it easier to create and use other drivers without code changes within VmShepherd itself, python's `entry point mechanism <https://packaging.python.org/specifications/entry-points/>`_ are used.
```
Comment (kwarunek) [inline docs/user/drivers/index.rst]: I think the note about entry_point is too detailed, plugins is enough for user perspective. On the other hand this nicely fits to development/plugins :)

Comment (evemorgen) [reply docs/user/drivers/index.rst]: Is it? I think it contains just enough details, it just introduces `entry point` term, which is being used later. Besides that change only hides link under text :P

Comment (kwarunek) [reply docs/user/drivers/index.rst]: So
- entry_points != plugins Plugins can be implemented with entry points like we did.
- entry_points in user docs can be confused with docker's entry point
- usage of vmshepherd does not require to understand advanced python packaging mechanism.
- the point of this paragraph is (AFAIU) to show that app is extendible (and it's easy), so where is the link to devel/plugins?

IMHO this sentence should be placed in development/plugins/index and here just link with some note about pluginability.

Comment (pkruk) [reply docs/user/drivers/index.rst]: entry poinst can be confusing with docker definition of entrypoint, well i think dev doc is a better place for this info ;)

Comment (evemorgen) [reply docs/user/drivers/index.rst]: fine

**Pull request merged**